### PR TITLE
feat(restapi): remove base_url in restapi connector

### PR DESCRIPTION
**Pull request by donch1989**
Because

- We should put the full URL in component setting, so that the user can have a better understanding of the meaning of the Restapi component.

This commit

- Move the URL field to the component configuration section, and only keep Authentication in the connector configuration.

### CODE REVIEW

File: pkg/restapi/client.go [hunk 1]
```diff
@@ -17,8 +17,8 @@ type TaskOutput struct {
 	Header     map[string][]string    `json:"header"`
 }
 
-func newClient(config *structpb.Struct, logger *zap.Logger) (*httpclient.Client, error) {
-	c := httpclient.New("REST API", getBaseURL(config),
+func newClient(baseUrl string, config *structpb.Struct, logger *zap.Logger) (*httpclient.Client, error) {
+	c := httpclient.New("REST API", baseUrl,
 		httpclient.WithLogger(logger),
 	)
```
Comment (jvallesm) [inline pkg/restapi/client.go]: We can also remove the `baseURL` parameter. `resty`'s `Execute` method can be called with an absolute URL if the `BaseURL` method is unset. I don't know if passing an empty string would be enough or if we'd need to modify any of the fields or call any of the functions in `c` but it'd be easy to find out
```suggestion
func newClient(config *structpb.Struct, logger *zap.Logger) (*httpclient.Client, error) {
	c := httpclient.New("REST API", "",
		httpclient.WithLogger(logger),
	)
```

Comment (donch1989) [reply pkg/restapi/client.go]: Oh, yeah, empty string can work, let me update it

Comment (donch1989) [reply pkg/restapi/client.go]: updated

File: pkg/restapi/connector_test.go [hunk 2]
```diff
@@ -157,9 +163,9 @@ func TestConnector_Test(t *testing.T) {
 	defID := uuid.Must(uuid.NewV4())
 
 	c.Run("nok", func(c *qt.C) {
-		got, err := connector.Test(defID, cfg("http://no-such.host", noAuthType), logger)
-		c.Check(err, qt.IsNotNil)
-		c.Check(got, qt.Equals, pipelinePB.Connector_STATE_ERROR)
+		got, err := connector.Test(defID, cfg(noAuthType), logger)
```
Comment (jvallesm) [inline pkg/restapi/connector_test.go]: We can remove this test case now

**Review by jvallesm — DISMISSED** [review]

File: pkg/restapi/connector_test.go (hunk 2 above)
Comment (donch1989) [reply pkg/restapi/connector_test.go]: done

**Pull request merged**